bad error handling when -s file is missing

**Describe the bug**
```
jbang -s jackson.jsh -i
[jbang] [ERROR] Cannot invoke "dev.jbang.source.ResourceRef.getOriginalResource()" because "resourceRef" is null
[jbang] Run with --verbose for more details
```

should report its jackson.jsh that is missing.

--- a/src/main/java/dev/jbang/source/RunContext.java
+++ b/src/main/java/dev/jbang/source/RunContext.java
@@ -412,7 +412,7 @@ public class RunContext {
 		Function<String, String> propsResolver = it -> PropertiesValueResolver.replaceProperties(it,
 				getContextProperties());
 		return sources	.stream()
-						.map(resolver::resolve)
+						.map(s -> resolveChecked(resolver, s))
 						.map(ref -> Source.forResourceRef(ref, propsResolver))
 						.collect(Collectors.toList());
 	}
@@ -424,14 +424,7 @@ public class RunContext {
 	}
 
 	public Code forResource(String resource) {
-		ResourceRef resourceRef = getResourceResolver().resolve(resource);
-
-		// Support URLs as script files
-		// just proceed if the script file is a regular file at this point
-		if (resourceRef == null || !resourceRef.getFile().canRead()) {
-			throw new ExitException(BaseCommand.EXIT_INVALID_INPUT,
-					"Script or alias could not be found or read: '" + resource + "'");
-		}
+		ResourceRef resourceRef = resolveChecked(getResourceResolver(), resource);
 
 		if (resourceRef instanceof AliasResourceResolver.AliasedResourceRef) {
 			// The resource we found was obtained from an alias which might
@@ -471,6 +464,17 @@ public class RunContext {
 		// note script file must be not null at this point
 		setOriginalRef(resource);
 		return forResourceRef(resourceRef);
+	}
+
+	private static ResourceRef resolveChecked(ResourceResolver resolver, String resource) {
+		ResourceRef ref = resolver.resolve(resource);
+		// Support URLs as script files
+		// just proceed if the script file is a regular file at this point
+		if (ref == null || !ref.getFile().canRead()) {
+			throw new ExitException(BaseCommand.EXIT_INVALID_INPUT,
+					"Script or alias could not be found or read: '" + resource + "'");
+		}
+		return ref;
 	}
 
 	public Code forFile(File resourceFile) {

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -1834,4 +1834,20 @@ public class TestRun extends BaseTest {
 					"Error transferring file: dummyrepo from http://dummyrepo/dummygroup/dummyart/0.1/dummyart-0.1.pom"));
 		}
 	}
+
+	@Test
+	void testMissingSource() throws IOException {
+		CommandLine.ParseResult pr = JBang.getCommandLine().parseArgs("run", "-s", "missing.jsh", "-i");
+		Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+		RunContext ctx = run.getRunContext();
+		try {
+			SourceSet ss = (SourceSet) ctx.forResourceRef(LiteralScriptResourceResolver.stringToResourceRef(null, ""));
+		} catch (ExitException ex) {
+			StringWriter sw = new StringWriter();
+			ex.printStackTrace(new PrintWriter(sw));
+			assertThat(sw.toString(), containsString(
+					"Script or alias could not be found or read: 'missing.jsh'"));
+		}
+	}
 }

